drop-in/src/main/java/com/adyen/checkout/dropin/ui/viewmodel/PaymentMethodsListViewModel.kt

```diff
@@ -8,227 +8,31 @@
-import androidx.lifecycle.LiveData
-import androidx.lifecycle.MutableLiveData
-import com.adyen.checkout.components.ComponentAvailableCallback
-import com.adyen.checkout.components.base.Configuration
+import androidx.lifecycle.ViewModel
+import androidx.lifecycle.viewModelScope
 import com.adyen.checkout.components.model.paymentmethods.PaymentMethod
-import com.adyen.checkout.components.model.paymentmethods.StoredPaymentMethod
-import com.adyen.checkout.components.model.payments.Amount
-import com.adyen.checkout.components.util.CurrencyUtils
-import com.adyen.checkout.components.util.PaymentMethodTypes
-import com.adyen.checkout.core.exception.CheckoutException
-import com.adyen.checkout.core.log.LogUtil
-import com.adyen.checkout.core.log.Logger
-import com.adyen.checkout.dropin.DropInConfiguration
-import com.adyen.checkout.dropin.R
-import com.adyen.checkout.dropin.checkPaymentMethodAvailability
-import com.adyen.checkout.dropin.ui.order.OrderModel
-import com.adyen.checkout.dropin.ui.paymentmethods.GiftCardPaymentMethodModel
-import com.adyen.checkout.dropin.ui.paymentmethods.PaymentMethodHeader
 import com.adyen.checkout.dropin.ui.paymentmethods.PaymentMethodListItem
-import com.adyen.checkout.dropin.ui.paymentmethods.PaymentMethodModel
-import com.adyen.checkout.dropin.ui.paymentmethods.PaymentMethodNote
-import com.adyen.checkout.dropin.ui.paymentmethods.StoredPaymentMethodModel
-import com.adyen.checkout.dropin.ui.stored.makeStoredModel
+import kotlinx.coroutines.flow.MutableStateFlow
+import kotlinx.coroutines.flow.StateFlow
+import kotlinx.coroutines.flow.launchIn
+import kotlinx.coroutines.flow.onEach
 
-@Suppress("TooManyFunctions")
-class PaymentMethodsListViewModel(
-    application: Application,
-    private val paymentMethods: List<PaymentMethod>,
-    storedPaymentMethods: List<StoredPaymentMethod>,
-    private val order: OrderModel?,
-    private val dropInConfiguration: DropInConfiguration,
-    private val amount: Amount
-) : AndroidViewModel(application), ComponentAvailableCallback<Configuration> {
+internal class PaymentMethodsListViewModel constructor(private val delegate: PaymentMethodsListDelegate) : ViewModel() {
 
-    private val paymentMethodsMutableLiveData: MutableLiveData<List<PaymentMethodListItem>> = MutableLiveData()
-    val paymentMethodsLiveData: LiveData<List<PaymentMethodListItem>> = paymentMethodsMutableLiveData
-
-    private var availabilitySum = 0
-    private var availabilitySkipSum = 0
-    private var availabilityChecksum = 0
-
-    private val storedPaymentMethodsList = mutableListOf<StoredPaymentMethodModel>()
-    private val paymentMethodsList = mutableListOf<PaymentMethodModel>()
-    private val orderPaymentMethodsList: List<GiftCardPaymentMethodModel> = setupOrderPaymentMethods(order)
+    private val paymentMethodsMutableFlow = MutableStateFlow<List<PaymentMethodListItem>>(emptyList())
+    internal val paymentMethodsFlow: StateFlow<List<PaymentMethodListItem>> = paymentMethodsMutableFlow
```

Comment:
Why use this structure? It's better to directly pass the delegate's flow like this:
```Kotlin
val paymentMethodsFlow: Flow<List<PaymentMethodListItem>> = delegate.paymentMethodsFlow
```